### PR TITLE
Adding webloader

### PR DESCRIPTION
**Pull request by tfoster**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,65 +1,50 @@
...
-		let cssAsset;
-	
-		const links = document.getElementsByTagName('link');
-
-		for (let i = 0; i < links.length; i++) {
-			const link = links[i];
-			if (link.rel === 'stylesheet' && link.href && link.href.indexOf('evolv.ai') >= 0 && link.href.indexOf('assets.css') >= 0) {
-				cssAsset = link;
-				break;
+		const styleSheets = document.styleSheets;
```
Comment (FrazerBayley) [inline src/index.js]: I hesitated using this api bc MDN named it experimental. The browser support seemed to be extensive but it wasnt as extensive as `document.scripts`. https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/styleSheets

File: src/index.js [hunk 2]
```diff
@@ -1,65 +1,50 @@
...
-		let jsAsset;
-	
-		const scripts = document.getElementsByTagName('script');
+		const scripts = document.scripts;
 	
 		for (let i = 0; i < scripts.length; i++) {
 			const script = scripts[i];
 			if (script.src && script.src.indexOf('evolv.ai') >= 0 && script.src.indexOf('assets.js') >= 0) {
-				jsAsset = script;
-				break;
```
Comment (FrazerBayley) [inline src/index.js]: Is break not supported in some browsers?

File: src/index.js [hunk 3]
```diff
@@ -1,65 +1,50 @@
 import Evolv from '@evolv/javascript-sdk';
```
Comment (FrazerBayley) [inline src/index.js]: If the client is being passed in now can we remove this import?

File: src/webloader.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+function injectStylesheet(endpoint, env, uid) {
+	const stylesheet = document.createElement('link');
+	stylesheet.setAttribute('rel', 'stylesheet');
+	stylesheet.setAttribute('type', 'text/css');
+	stylesheet.setAttribute('href', endpoint + '/' + env + '/' + uid + '/assets.css');
+
+	document.head.appendChild(stylesheet);
+}
+
+if (!window.evolv) {
```
Comment (FrazerBayley) [inline src/webloader.js]: I had a question about where we should do this. When i was using the document.getElementsByTagName() api and the asset.js was lower than the AssetManager on the page, the api could not find the script. So unless the other apis (document.scripts) can find a lower script it almost seems like we need to require that asset.js be higher than the AssetManager. At that point should the asset.js set up the evolv object on the window?

File: src/index.js [hunk 5]
```diff
@@ -117,21 +103,14 @@ function main(client, options) {
 	});
 }
 
-function EvolvAssetManager(options) {
-	const client = new Evolv(options);
+function EvolvAssetManager(client) {
+	client.context.set('web.url', window.location.href);
 
 	// Expose client and context proprties
 	Object.defineProperty(this, 'client', { get: function () { return client }});
```
Comment (FrazerBayley) [inline src/index.js]: Do we still need to do this now that we're attaching the client to the window?

File: src/index.js [hunk 6]
```diff
@@ -117,21 +103,14 @@ function main(client, options) {
 	});
 }
 
-function EvolvAssetManager(options) {
-	const client = new Evolv(options);
+function EvolvAssetManager(client) {
```
Comment (FrazerBayley) [inline src/index.js]: If we're attaching the client to the window whats the reasoning behind passing it in here instad of just accessing it globally?

File: src/index.js (hunk 5 above)
Comment (tfoster) [reply src/index.js]: Yeah, the client shouldn't collect any attributes, the AssetManager should pass what it believes is relevant.

File: src/index.js (hunk 6 above)
Comment (tfoster) [reply src/index.js]: This won't only be consumed by the webloader. It will also be instantiated directly. It can't assume it has access to it globally.

**Pull request merged**